project/src/com/google/daggerquery/protobuf/BUILD

```diff
@@ -12,12 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-load("@rules_java//java:defs.bzl", "java_library")
-
 package(default_visibility = ["//visibility:public"])
 
-java_library(
-    name = "dagger_query_protobuf",
-    srcs = glob(["autogen/*.java"]),
-    deps = ["//third_party/java/protobuf:protobuf"],
+java_proto_library(
+    name = "binding_graph_java_proto",
+    deps = [
+        ":binding_graph_proto",
+        ":dependency_proto"
```

Comment:
I suspect `:dependency_proto` may not be required. Would you mind trying to remove and test if it continues to build?

Reply:
You are totally right. I tested it out, and this dependency is only needed for the proto_library rule.